koa_crud/src/data-access/database/index.ts

```diff
@@ -0,0 +1,19 @@
+import mongoose from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+
+const mongod = new MongoMemoryServer();
+
+export const initializeDatabase = async (): Promise<void> => {
+  const uri = await mongod.getConnectionString();
+
+  const mongooseOpts = {
+    useNewUrlParser: true,
+    autoReconnect: true,
+    reconnectTries: Number.MAX_VALUE,
+    reconnectInterval: 1000,
+  };
+
+  await mongoose.connect(uri, mongooseOpts);
+
+  mongoose.set('runValidators', true);
+};
```

Comment:
You should add another setup for mongodb in-memory just for testing. You should leave your previous set up as is.

In memory mongodb is only for testing.